src/stdlib_bitsets_large.fypp

```diff
@@ -91,12 +91,11 @@ contains
 
     pure module subroutine assign_large( set1, set2 )
 !     Used to define assignment for bitset_large
-        type(bitset_large), intent(out) :: set1
-        type(bitset_large), intent(in)  :: set2
+        type(bitset_large), intent(inout) :: set1
+        type(bitset_large), intent(in)    :: set2
 
         set1 % num_bits = set2 % num_bits
-        allocate( set1 % blocks( size( set2 % blocks, kind=bits_kind ) ) )
-        set1 % blocks(:) = set2 % blocks(:)
+        set1 % blocks = set2 % blocks(:)
```

Comment:
This might restrict `stdlib`  to the most updated compilers. Why did you change it?